frontend/src/citizen-frontend/calendar/ChildSelector.tsx

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2017-2024 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { useMemo } from 'react'
+import styled, { useTheme } from 'styled-components'
+
+import { getDuplicateChildInfo } from 'citizen-frontend/utils/duplicated-child-utils'
+import { BoundFormState, useBoolean } from 'lib-common/form/hooks'
+import { ReservationChild } from 'lib-common/generated/api-types/reservations'
+import { formatFirstName } from 'lib-common/names'
+import { UUID } from 'lib-common/types'
+import { SelectionChip } from 'lib-components/atoms/Chip'
+import { StatusIcon } from 'lib-components/atoms/StatusIcon'
+import {
+  FixedSpaceColumn,
+  FixedSpaceFlexWrap
+} from 'lib-components/layout/flex-helpers'
+import { defaultMargins } from 'lib-components/white-space'
+import { fasExclamationTriangle } from 'lib-icons'
+
+import { useTranslation } from '../localization'
+
+interface ChildSelectorProps {
+  childItems: ReservationChild[]
+  bind: BoundFormState<UUID[]>
+}
+
+export default React.memo(function ChildSelector({
+  childItems,
+  bind
+}: ChildSelectorProps) {
+  const t = useTranslation()
+
```

Comment:
Vältä `document.getElementById` käyttöä blur eventissä. Sen sijaan voit käyttää React referenssiä:
```suggestion

  const chipGroupContainerRef = React.useRef<HTMLDivElement>(null)
```